hngprojects/hng_boilerplate_nestjs
# feat: implement subscription count retrieval service method and tests

## PR Description

### Issue Description

**What does the PR do?**
This PR implements a new service method in the `NewsletterService` to retrieve the subscription count and adds corresponding unit tests to ensure its functionality.

**How should this be manually tested?**
1. Clone the repository and checkout the `feat/implement-subscription-count-service` branch.
2. Run `npm install` to install all dependencies.
3. Use `npm run test` to execute the unit tests.
4. Verify that all tests pass, particularly the test for the subscription count retrieval method.

**Successful Response**

```json
{
  "message": "Subscription count fetched successfully",
  "data": {
    "subscription_count": 5
  }
}
```

**Authentication Error**

If the request is not authenticated, you should see an error response similar to:

```json
{
  "statusCode": 401,
  "message": "Unauthorized"
}
```

**Invalid UserID Error**

This PR does not involve user ID validation as it pertains to subscription counts without user-specific data. No such errors should occur in this context.

**Validation**

Validation errors do not apply directly to this feature as it involves a straightforward retrieval process without user inputs.

**Task**
- Implement a service method to retrieve and return the subscription count.
- Write unit tests to ensure the method works correctly.

**Reference Issue**
None

### Screenshots
Screenshots for proof of testing (such as Postman screenshots) will be provided here, showing:
1. Successful subscription count retrieval.

![image](https://github.com/user-attachments/assets/b3eb014f-58b0-4040-93ee-46e5926c5ab8)


2. Authentication errors if applicable.

![image](https://github.com/user-attachments/assets/b4b15374-b847-4f4a-b612-1075337a1d51)

5. Any other relevant testing proof.

### Code Quality and Best Practices
6. All comments and `console.log` statements have been removed.
7. No unnecessary `try/catch` blocks for internal server errors, relying on the existing interceptor for error handling.
8. Used a constant for the success message, stored in `/helpers/SystemMessages`.
9. Avoided redundant method implementations by leveraging existing methods for user detail fetching where applicable.
